packages/client/libclient-py/quickmpc/utils/parse_csv.py

```diff
@@ -41,29 +82,103 @@ def to_float(val: str) -> float:
         return val_float
 
 
-def parse(data: List[List[str]]) \
-        -> Tuple[List[List[float]], List[str]]:
-    schema: List[str] = data[0]
-    secrets: List[List[float]] = [
-        [to_float(x) for x in row] for row in data[1:]]
+def to_int(val: str, encoding='utf-8') -> int:
+    encoded = val.encode(encoding)
+    return int.from_bytes(encoded, byteorder='big')
+
+
+def check_float_data(val: str) -> bool:
+    try:
+        _ = float(val)
+        return True
+    except ValueError:
+        return False
+
+
+def find_type(col_schema: str,
+              col_data: List[str], is_matching_column: bool) \
+        -> ShareValueTypeEnum.ValueType:
+    # check tag
+    type_str, *remains = col_schema.split(':')
+
+    if len(remains) > 0:
+        if type_str in SUPPORT_TAGS:
+            return SUPPORT_TAGS[type_str]
+
+    # check if the column of interest is the one used for matching
+    if is_matching_column:
+        return ShareValueTypeEnum.SHARE_VALUE_TYPE_FIXED_POINT
+
+    # determine type through number of string which is considered float value
+    res_type = ShareValueTypeEnum.SHARE_VALUE_TYPE_FIXED_POINT
+    num_float = sum([check_float_data(v) for v in col_data])
+
+    if num_float < len(col_data) / 2:
```

Comment:
- We're worried about the judgment that it's less than half
- If there is at least one string, it is better to treat it as a string column
   - because it is impossible to calculate